### ISSUE TITLE
MySam plugins catalog

### ISSUE THREAD

**Issue by mabrasil** [daffl/mysam#4]
Given some facts like:

- The great number of possibilities that MySan can achieve by plugins
- The ease of creating new plugins and publishing them to *npm*
- The growing popularity of this project

I'd bet that the number of MySam plugins will grow very fast. Maybe it would be cool to have a place to centralize them. 

@thebergamo proposed in #3 the creation of a file or a section in the `README` and I think that this would be also important - but only for *official* (core-team developed) plugins - that's why I opened another issue instead of just commenting the other one.

Many tools in the *node.js world* have their own extensions catalog; e.g: automation tools, like [gulp's  plugins registry](http://gulpjs.com/plugins/), preprocessors, like [elm's catalog](http://package.elm-lang.org/) and so on.

Maybe now it's not a needing, but who knows in a near feature? If you don't have time to create one, I'd suggest, for now, include in the [plugins creation guidelines](https://github.com/daffl/mysam#creating-plugins) instructions to any MySam plugin have a tag like `mysam-plugin`; e.g:

```json
{
  "name": "mysam-myplugin",
  "main": "server",
  "mysam": {
    "client": "frontend",
    "styles": "styles.css"
  },
  "keywords": [
    "my-keyword-1",
    "my-keyword-2",
    "my-keyword-3",
    "mysam-plugin"
  ]
}
```
In short time it would at least allow you doing something like [broccolijs](http://broccolijs.com/#) does - which is linking to a npm *seach-by-keyword* result. In this case you would have a [`https://www.npmjs.com/browse/keyword/mysam-plugin`](https://www.npmjs.com/browse/keyword/broccoli-plugin) link in the project's homepage.

In the long-term, you can use the idea of a special keyword in the `package.json` to create - or fork - something like [Custom Elements](https://customelements.io/) - which would search for published packages with the `mysam-plugin` keyword and gather them in a beautiful and *easy-to-publish* registry.

BTW, I loved the project :heart:

**Comment by thebergamo** [comment]
@mabrasil I think this is the right way! This project is awesome and will grown fast =D

**Comment by IgorAntun** [comment]
+1, that sounds great.

**Comment by daffl** [comment]
This is great feedback. A website is definitely in the works and it will have a list of plugins based on the NPM tag. I'm also looking into creating a [Yeoman](http://yeoman.io/) generator for new plugins that initialize everything similar to what is described in the readme so that we can hopefully just do `mysam create plugin`.

Can someone add the suggested link to https://www.npmjs.com/browse/keyword/mysam-plugin to the Readme and submit a pull request? I'll re-publish the weather plugin (currently the only one I am aware of) with the keyword and once I am done with the todos list there will at least be two showing up already :)

**Comment by mabrasil** [comment]
Hey, it's great to know that you've been plannign this! 

> I'm also looking into creating a Yeoman generator for new plugins that initialize everything similar to what is described in the readme so that we can hopefully just do mysam create plugin.

That one would be my next issue. I think that *issues* are a very good way of organizing the project's *TODOs*. And I think it's worth opening one to discuss this generator.

Another good issue would be one to purpose a logo to this project.

If you agree, I may open both of them.

> Can someone add the suggested link to https://www.npmjs.com/browse/keyword/mysam-plugin to the Readme and submit a pull request?

I've done this and also included some small docs suggestions in #6. Hope you like it :smiley:

**Comment by KarimJedda** [comment]
I love this thing, here's a logo proposal, hope you like it http://imgur.com/RLnDOf2, i can work on the resolution/sharpness if you want :)

**Comment by daffl** [comment]
@mabrasil Thanks for the pull request. It is merged and looking good.
@KarimJedda awesome! I would hold off a little so that we can plan what the next steps in regards to design and homepage should be but I'll definitely keep you in the loop.

**Comment by thejase** [comment]
This is fantastic!  Whatever you do, do /not/ sell the rights to this property!!
